### PR TITLE
[5] [RFC] [cli] extension Update command

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #40693 .

### Summary of Changes

if you want an extension to be installable from cli then add this tag `<updatefromcli>true</updatefromcli>` to the updateserver.xml
for example 
image

new command to perform the extension updated 
`php cli\joomla.php extension:update --eid=extension_id`

### Testing Instructions

run `php cli\joomla.php extension:update --eid=extension_id`

### Actual result BEFORE applying this Pull Request
N/A

### Expected result AFTER applying this Pull Request

run
`php cli\joomla.php extension:update --eid=249`
image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Hackwar** [comment]
I would strongly advise against this at this moment. We should first introduce a flag into extension manifests to mark if an extension supports updating via CLI, because otherwise especially the big extensions with their own install scripts will all fail when updating.

**Comment by alikon** [comment]
doesn't that will fail from backend com_installer as well ?

**Comment by Hackwar** [comment]
No, an extension which does redirects after installation, maybe does ajax calls when the update is finished or expects a web context in generell, will fail with the update when you are in a CLI context.

**Comment by alikon** [comment]
that's is more like a general joomla issue
not all joomla api performs like a client agnostic 
there already are plenty of issue reports about that 
cli vs workflows just as a recent example

out of scope of this silly pr

**Comment by Hackwar** [comment]
Considering that this could pretty much destroy a website, I don't consider this something we should just risk. Yes, there are things which don't work right now, but that isn't a reason to introduce yet something else which is broken.

**Comment by alikon** [comment]
the `update:extensions:check` pr #42844

File: libraries/src/Console/ExtensionUpdateCommand.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    /**
+     * Transforms extension arrays into required form
+     *
+     * @param   array  $extensions  Array of extensions
+     *
+     * @return array
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    protected function getExtensionsNameAndId($extensions): array
```
Comment (richard67) [inline libraries/src/Console/ExtensionUpdateCommand.php]: Shouldn't this one be removed from this PR here after the split of PR #42844 ?

Comment (alikon) [reply libraries/src/Console/ExtensionUpdateCommand.php]: Is needed if we want to show more info about the update from to etc...

image

Comment (richard67) [reply libraries/src/Console/ExtensionUpdateCommand.php]: But hasn't that method been renamed in the other PR?

Comment (alikon) [reply libraries/src/Console/ExtensionUpdateCommand.php]: yes
i'll align

**Comment by alikon** [comment]
i've introduced a tag `<updatefromcli>true</updatefromcli>`  for the updateserver.xml

**Comment by alikon** [comment]
no interest

**Pull request closed without merging**